pkg/providers/version/version.go

```diff
@@ -69,32 +70,26 @@ func NewDefaultProvider(kubernetesInterface kubernetes.Interface, cache *cache.C
 }
 
 func (p *DefaultProvider) Get(ctx context.Context) (string, error) {
-	var version string
-	var versionSource string
 	if version, ok := p.cache.Get(kubernetesVersionCacheKey); ok {
 		return version.(string), nil
 	}
-	output, err := p.eksapi.DescribeCluster(ctx, &eks.DescribeClusterInput{
-		Name: lo.ToPtr(options.FromContext(ctx).ClusterName),
-	})
-	if err != nil {
-		if !awserrors.IsAccessDenied(err) {
-			return "", err
-		}
-		output, err := p.kubernetesInterface.Discovery().ServerVersion()
-		if err != nil {
-			return "", fmt.Errorf("getting kubernetes version from the kubernetes API")
-		} else if output != nil {
-			version = fmt.Sprintf("%s.%s", output.Major, strings.TrimSuffix(output.Minor, "+"))
-			versionSource = "Kubernetes API"
-		}
-	} else if lo.FromPtr(output.Cluster.Version) != "" {
-		version = *output.Cluster.Version
-		versionSource = "EKS DescribeCluster"
-	} else {
-		return "", fmt.Errorf("unable to retrieve Kubernetes version from EKS DescribeCluster")
+
+	var version, versionSource string
+	var err error
+
+	if options.FromContext(ctx).EKSControlPlane {
+		version, versionSource, err = p.getEKSVersion(ctx)
 	}
+	if !options.FromContext(ctx).EKSControlPlane {
+		version, versionSource, err = p.getK8sVersion()
+	}
+	if version == "" && err != nil {
+		version, versionSource = *p.previousVersion.Clone(), *p.previousVersionSource.Clone()
+	}
+
 	p.cache.SetDefault(kubernetesVersionCacheKey, version)
```

Comment:
I don't think you need a cache at all with this logic change that you made -- really, what you need is to store the last time that you updated and the version data that you have -- if you are within 5 minutes or something of your last update time, then return what you have; otherwise, try to update the data